.github/workflows/create-pull-request.yml

```diff
@@ -0,0 +1,46 @@
+name: Auto Create Pull Request
+
+on:
+  workflow_call:
+    inputs:
+      base_branch:
+        description: "The base branch to merge the pull request"
+        required: false
+        type: string
+        default: main
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  create-pr:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Create PR Title
+        id: pr-title
+        run: |
+          branch_name=${GITHUB_REF_NAME}
+
+          # get pr description with issue number
+          if [[ ${branch_name} =~ i([0-9]+) ]]; then
+            issue_number="${BASH_REMATCH[1]}"
+            branch_name=$(echo ${branch_name} | sed 's/i[0-9]*_//')
+            echo "PR_DESCRIPTION=Resolves #${issue_number}" >> $GITHUB_ENV
+          else
+            echo "PR_DESCRIPTION=No issue reference found" >> $GITHUB_ENV
+          fi
+
+          # format pr title from branch name (capitalization, replace dashes with spaces)
+          echo "PR_TITLE=$(echo ${branch_name} | sed 's/\b\(.\)/\u\1/g' | sed 's/-/ /g')" >> $GITHUB_ENV
+
+          echo "BRANCH_NAME=${branch_name}" >> $GITHUB_ENV
+      - name: Create Pull Request
+        uses: thomaseizinger/create-pull-request@1.3.1
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          title: ${{ env.PR_TITLE }}
+          body: ${{ env.PR_DESCRIPTION }}
+          base: ${{ inputs.base_branch }}
+          head: ${{ github.ref }}
```

Comment:
We somehow commited during refinement, that we want to try to avoid external/untrusted actions. Would is be hard to to this as a rest request?